server/app/controllers/admin/ProgramAdminManagementController.java

```diff
@@ -52,34 +52,39 @@ public Result edit(Http.Request request, long programId) {
     return this.loadProgram(request, programId, Optional.empty());
   }
 
-  /**
-   * Promotes the given account emails to the role of PROGRAM_ADMIN. If there are errors, return a
-   * redirect with flashing error message.
-   */
+  /** Removes `adminEmail` as a program admin for the program identified by `programId`. */
   @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
-  public Result update(Http.Request request, long programId) {
-    Form<ManageProgramAdminsForm> form = formFactory.form(ManageProgramAdminsForm.class);
-    if (form.hasErrors()) {
-      return badRequest();
+  public Result delete(Http.Request request, long programId) {
+    Form<ManageProgramAdminsForm> form =
+        formFactory.form(ManageProgramAdminsForm.class).bindFromRequest(request);
+    try {
+      roleService.removeProgramAdmins(programId, ImmutableSet.of(form.get().getAdminEmail()));
+      return redirect(routes.ProgramAdminManagementController.edit(programId));
+    } catch (ProgramNotFoundException e) {
+      return notFound(e.getLocalizedMessage());
     }
-    ManageProgramAdminsForm manageAdminForm = form.bindFromRequest(request).get();
+  }
 
+  /** Adds a new admin email. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result add(Http.Request request, long programId) {
+    Form<ManageProgramAdminsForm> form =
+        formFactory.form(ManageProgramAdminsForm.class).bindFromRequest(request);
+    if (Strings.isNullOrEmpty(form.get().getAdminEmail())) {
```

Comment:
I think we should trim the email in case there's whitespace.

Reply:
Done